auth_mellon_config.c

```diff
@@ -105,6 +105,10 @@ static const int default_env_vars_count_in_n = -1;
 /* The default list of trusted redirect domains. */
 static const char * const default_redirect_domains[] = { "[self]", NULL };
 
+/* The default signature method used to sign messages */
+static const LassoSignatureMethod default_signature_method =
+    LASSO_SIGNATURE_METHOD_RSA_SHA256;
```

Comment:
This will not really work with inheritance and overrides. By using a default value that matches a value the user can set, you cannot tell the difference between a value overridden by the user and a value the user hasn't set.

Reply:
I must be missing something, this seems consistent with how the existing config code works. Is there some value in detecting the case when someone explicitly set a value to the default? If so isn't there quite a few other values that suffer from the same issue?

Reply:
I haven't enforced the rule consistently, so there are a few that are not behaving properly. However, most should have a way to override the option back to its default in the case where it has been set to a different value. I would prefer to not add more of the options that have inconsistent override rules if possible.

For options that boil down to an integer, the `CFG_MERGE` and `CFG_VALUE` macros can help: https://github.com/UNINETT/mod_auth_mellon/blob/v0.13.1/auth_mellon.h#L336-L366